Packs/FeedMandiant/Integrations/FeedMandiant/FeedMandiant.py

```diff
@@ -0,0 +1,655 @@
+        super().__init__(base_url=base_url, auth=(username, password), verify=verify, proxy=proxy, ok_codes=[200])
+        self._headers = {
+            'X-App-Name': "content.xsoar.cortex.paloaltonetworks.v1.0",
+            'Accept': 'application/json',
+            'Authorization': f'Bearer {self._get_token()}'
+        }
+        self.timeout = timeout
+        self.first_fetch = first_fetch
+        self.limit = limit
+        self.types = types
+        self.metadata = metadata
+        self.tlp_color = tlp_color
+        self.tags = tags
+        self.enrichment = enrichment
+
+        LOG.add_replace_strs(self._get_token())
+
+    def _get_token(self) -> str:
+        """
+        Obtains token from integration context if available and still valid.
+        After expiration, new token are generated and stored in the integration context.
+        Returns:
+            str: token that will be added to authorization header.
+        """
+        integration_context = get_integration_context()
+        token = integration_context.get('token', '')
+        valid_until = integration_context.get('valid_until')
+
+        now_timestamp = arg_to_datetime('now').timestamp()  # type:ignore
+        # if there is a key and valid_until, and the current time is smaller than the valid until
+        # return the current token
+        if token and valid_until:
+            if now_timestamp < valid_until:
+                return token
+
+        # else generate a token and update the integration context accordingly
+        token = self._generate_token()
+
+        return token
+
+    def _generate_token(self) -> str:
+        """
+        Generates new token.
+        """
+        data = {
+            'grant_type': 'client_credentials'
+        }
+        resp = self._http_request(method='POST', url_suffix='token', resp_type='json', data=data)
+        self._token = resp.get('access_token')
+
+        integration_context = get_integration_context()
+        integration_context.update({'token': self._token})
+        # Add 10 minutes buffer for the token
+        integration_context.update({'valid_until': datetime.timestamp(datetime.now(timezone.utc)) - 600})
+        set_integration_context(integration_context)
+
+        return self._token
+
+    def get_indicator_additional_info(self, identifier: str, indicator_type: str, info_type: str = "") \
+            -> Union[Dict, List]:
+        """
+        Get additional information for given indicator.
+
+        Args:
+            identifier (Dict): Indicator's identifier.
+            indicator_type (str): The indicator type.
+            info_type (str): Type of additional info
+        Returns:
+            Dict: Additional metadata of the indicator.
+        """
+        url = f"v4/{MAP_TYPE_TO_URL[indicator_type]}"
+        url = urljoin(url, identifier)
+        url = urljoin(url, info_type)
+        if url[-1] == '/':
+            url = url[:-1]
```

Comment:
Why would you have a `/` at the end?

Reply:
urljoin would add ```/``` at the end if the second argument is empty